core/src/create.jl

```diff
@@ -201,6 +201,32 @@ end
 const nonconservative_nodetypes =
     Set{String}(["Basin", "LevelBoundary", "FlowBoundary", "Terminal", "User"])
 
+"""
+Get the chunk sizes for DiffCache; differentiation w.r.t. u
+and t (the latter only if a Rosenbrock algorithm is used).
+"""
+function get_chunk_sizes(config::Config, chunk_size::Int)::Vector{Int}
+    chunk_sizes = [chunk_size]
+    if Ribasim.config.algorithms[config.solver.algorithm] <:
+       OrdinaryDiffEqRosenbrockAdaptiveAlgorithm
+        push!(chunk_sizes, 1)
+    end
+    return chunk_sizes
+end
+
+"""
+If the tuple of variables contains a Dual variable, return the first one.
+Otherwise return the last variable.
+"""
+function get_diffvar(variables::Tuple)
```

Comment:
I think this would be a bit more idiomatic than using a Tuple.

```
julia> f(args...) = [a for a in args]
f (generic function with 1 method)

julia> f(3, 4)
2-element Vector{Int64}:
 3
 4
```